pkg/servicemesh/federation/discovery/controller.go

```diff
@@ -116,8 +116,12 @@ func (c *Controller) Run(stopChan <-chan struct{}) {
 	}
 }
 
+func (c *Controller) Start(stopChan <-chan struct{}) {
+	c.Controller.StartController(stopChan, c.HasSynced)
```

Comment:
Federation discovery controller is not able to call its own `HasSynced()` when it calls `Start()` of its embedded `controller.Controller`. Now it calls `Start()` of the embedded controller, but it passes its own `HasSync()`.

Reply:
But where is this `Start()` function being called?

Reply:
Here: https://github.com/maistra/istio/blob/403a6b63c620b8b7c2376f3a24b17022acbc0509/pkg/servicemesh/federation/federation.go#L175.